packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -163,10 +173,42 @@ library LibVerifying {
             proofTime = uint64(fc.provenAt - blk.proposedAt);
         }
 
-        uint64 reward;
+        TaikoData.Bid memory winningBid = state.auctions[LibAuction.blockIdToBatchId(config, blk.blockId)].bid;
+        
+        uint64 rewardPerBlock = fc.gasUsed * winningBid.feePerGas;
+    
+        state.avgRewardPerBlock = uint64(LibUtils.movingAverage(state.avgRewardPerBlock, rewardPerBlock, 100));
+
+        // Check if prover is equal to auction winner. If so, just distribute normall if not, distribute reward normall
+        // + half of the deposit / block
+        uint256 batchId = LibAuction.blockIdToBatchId(config, blk.blockId);
+        TaikoData.Auction memory auction = state.auctions[batchId];
+
+        TaikoToken tkoToken = TaikoToken(resolver.resolve("tko_token", false));
+        // Prover indeed the one who won the auction
+        if(auction.bid.prover == fc.prover) {
+            try tkoToken.transfer(auction.bid.prover, auction.bid.deposit / config.auctionBatchSize + rewardPerBlock) {}
+            catch {
+                // allow to fail in case they have a bad onTokenReceived
```

Comment:
I don't think it's a good idea to allow failure here. We controls the implementation of `onTokenReceived`, so no failure should be tolerated.